tests/foreman/api/test_provisioningtemplate.py

```diff
@@ -406,7 +404,65 @@ def test_positive_template_check_vlan_parameter(
                 },
             ],
         ).create()
-        provision_template = host.read_template(data={'template_kind': 'provision'})
-        assert f'interfacename=vlan{tag}' in str(provision_template)
-        ipxe_template = host.read_template(data={'template_kind': 'iPXE'})
-        assert f'vlan=vlan{tag}:{identifier}' in str(ipxe_template)
+        provision_template = host.read_template(data={'template_kind': 'provision'})['template']
+        assert f'interfacename=vlan{tag}' in provision_template
+        ipxe_template = host.read_template(data={'template_kind': 'iPXE'})['template']
+        assert f'vlan=vlan{tag}:{identifier}' in ipxe_template
+
+    @pytest.mark.parametrize('module_sync_kickstart_content', [7, 8, 9], indirect=True)
+    @pytest.mark.parametrize('pxe_loader', ['uefi'], indirect=True)
+    @pytest.mark.parametrize('boot_mode', ['Static', 'DHCP'])
+    def test_positive_template_subnet_with_boot_mode(
+        self,
+        module_sync_kickstart_content,
+        module_target_sat,
+        module_sca_manifest_org,
+        module_location,
+        default_architecture,
+        default_partitiontable,
+        pxe_loader,
+        boot_mode,
+    ):
+        """Check whether boot mode paremeter in subnet respected when PXELoader UEFI is used,
+            and properly rendered in the provisioning templates
+
+        :id: 2decc787-59b0-41e6-96be-5dd9371c8966
+
+        :expectedresults: templates should get render and contains boot mode used as set in subnet
+
+        :BZ: 2168967, 1955861, 1784012
+
+        :customerscenario: true
+        """
+        subnet = module_target_sat.api.Subnet(
+            name=gen_string('alpha'),
+            organization=[module_sca_manifest_org],
+            location=[module_location],
+            network='192.168.0.1',
+            mask='255.255.255.240',
+            boot_mode=boot_mode,
+        ).create()
```

Comment:
Shouldn't this be a fixture?

Reply:
it is surely part of setup for the test, but I need this in test as I'm parametrizing it rn. Also if we need this in other tests so will surely consider moving it to global fixtures

Reply:
Can't that be resolved by indirect parametrization?

Reply:
I just need dummy subnet, so I don't think any test might require that, so I don't think it'll make sense to move it to fixture